WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginNotification.java

```diff
@@ -0,0 +1,69 @@
+package org.wordpress.android.ui.accounts.login;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.BitmapFactory;
+import android.support.annotation.StringRes;
+import android.support.v4.app.NotificationCompat;
+
+import org.wordpress.android.R;
+import org.wordpress.android.util.AutoForeground;
+
+class LoginNotification {
+    private static Intent getPendingIntent(Context context) {
+        Intent resumeIntent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
+
+        // apparently, we need to null out the package name to make the intent resume the app
```

Comment:
wow didn't know this. Interesting! Do we have a reference or explanation as to why this is like that? Something we can put in here to explain the code a bit more

Reply:
That's something I noticed as the difference between `getLaunchIntentForPackage()` and the simple Intent I manually constructed ([like this](https://stackoverflow.com/a/5502950)). The Intent returned from `getLaunchIntentForPackage()` fails to resume the app with the current Task in it. I don't have a reference to this behaviour.

Reply:
Can we add a comment explaining that what this `getPendingIntent` actually tries to do is to recover the same intent that was launched in the first place, and uses it for making the Notification tap action "resume"  the original intent?

Reply:
Sure thing, makes sense. Did it with b8243bc.

Reply:
👍  I've been considering this for a while to check if there's an alternative, "cleaner" way of resuming the app (and also possibly have better code instead of a comment clarifying why we are practically defeating the purpose of calling `getLaunchIntent**ForPackage**` and then nullifying the intent's pacakage immediately after). I would have liked to use explicit intents but that also means we should be able to tell which `Activity` the user was on, thus making the whole process more convoluted. Just wanted to clarify why I thought a longer comment was good to have here.

Reply:
Totally, I'm there with you. I think that `getLaunchIntentForPackage()` had just one job to do but yeah, falls short apparently.